diff --git a/User/CAN/hy_can.c b/User/CAN/hy_can.c
index 09afad9..ca9391c 100644
--- a/User/CAN/hy_can.c
+++ b/User/CAN/hy_can.c
@@ -161,7 +161,62 @@ void CAN_IRQHandler()
 					s_cancom->get_charger_msg.input1_AC_voltage_x10V = INT8TO16(RXMsg.dataA[2], RXMsg.dataA[3]);
 
 					break;
+
+				case GW_MODULE_2_MSG_100MS_MSG_FRAME_ID://返回电压电流和状态
+					updatemodule(charger);
+				
+					s_cancom->get_charger_msg.output2_current_x10A = INT8TO16(RXMsg.dataB[2],RXMsg.dataB[3]);
+					s_cancom->get_charger_msg.output2_voltage_x10V = INT8TO16(RXMsg.dataB[0],RXMsg.dataB[1]);
+					s_cancom->get_charger_msg.statu2_1 = RXMsg.dataA[0];
+					s_cancom->get_charger_msg.statu2_2 = RXMsg.dataA[1];
+					s_cancom->get_charger_msg.temperature2_x1degree = RXMsg.dataA[3]-40;
+					
+					break;
+				case GW_MODULE_2_MSG_500MS_MSG_FRAME_ID:
+					updatemodule(charger);
+
+					s_cancom->get_charger_msg.battery2_voltage = INT8TO16(RXMsg.dataA[0], RXMsg.dataA[1]);
+					s_cancom->get_charger_msg.input2_AC_voltage_x10V = INT8TO16(RXMsg.dataA[2], RXMsg.dataA[3]);
+
+					break;
+
+
+				case GW_MODULE_3_MSG_100MS_MSG_FRAME_ID://返回电压电流和状态
+					updatemodule(charger);
+				
+					s_cancom->get_charger_msg.output3_current_x10A = INT8TO16(RXMsg.dataB[2],RXMsg.dataB[3]);
+					s_cancom->get_charger_msg.output3_voltage_x10V = INT8TO16(RXMsg.dataB[0],RXMsg.dataB[1]);
+					s_cancom->get_charger_msg.statu3_1 = RXMsg.dataA[0];
+					s_cancom->get_charger_msg.statu3_2 = RXMsg.dataA[1];
+					s_cancom->get_charger_msg.temperature3_x1degree = RXMsg.dataA[3]-40;
 					
+					break;
+				case GW_MODULE_3_MSG_500MS_MSG_FRAME_ID:
+					updatemodule(charger);
+
+					s_cancom->get_charger_msg.battery3_voltage = INT8TO16(RXMsg.dataA[0], RXMsg.dataA[1]);
+					s_cancom->get_charger_msg.input3_AC_voltage_x10V = INT8TO16(RXMsg.dataA[2], RXMsg.dataA[3]);
+
+					break;	
+
+				case GW_MODULE_4_MSG_100MS_MSG_FRAME_ID://返回电压电流和状态
+					updatemodule(charger);
+				
+					s_cancom->get_charger_msg.output4_current_x10A = INT8TO16(RXMsg.dataB[2],RXMsg.dataB[3]);
+					s_cancom->get_charger_msg.output4_voltage_x10V = INT8TO16(RXMsg.dataB[0],RXMsg.dataB[1]);
+					s_cancom->get_charger_msg.statu4_1 = RXMsg.dataA[0];
+					s_cancom->get_charger_msg.statu4_2 = RXMsg.dataA[1];
+					s_cancom->get_charger_msg.temperature4_x1degree = RXMsg.dataA[3]-40;
+					
+					break;
+				case GW_MODULE_4_MSG_500MS_MSG_FRAME_ID:
+					updatemodule(charger);
+
+					s_cancom->get_charger_msg.battery4_voltage = INT8TO16(RXMsg.dataA[0], RXMsg.dataA[1]);
+					s_cancom->get_charger_msg.input4_AC_voltage_x10V = INT8TO16(RXMsg.dataA[2], RXMsg.dataA[3]);
+
+					break;
+
 
 				//千航
 				case 0x180215f4:
@@ -318,10 +373,10 @@ uint32_t hy_can_get_output_current_x10A(void){
 
 
 
-		return s_cancom->get_charger_msg.output1_current_x10A;//+
-//		s_cancom->get_charger_msg.output2_current_x10A+
-//		s_cancom->get_charger_msg.output3_current_x10A+
-//		s_cancom->get_charger_msg.output4_current_x10A;
+		return s_cancom->get_charger_msg.output1_current_x10A+
+		s_cancom->get_charger_msg.output2_current_x10A+
+		s_cancom->get_charger_msg.output3_current_x10A+
+		s_cancom->get_charger_msg.output4_current_x10A;
 
 
 }
@@ -343,11 +398,10 @@ uint32_t hy_can_get_charger_module_temperature_x10degree(void){
 uint8_t hy_can_get_charger_module_statu1(void){
 
 
-			return s_cancom->get_charger_msg.statu1_1;
-//				|
-//					s_cancom->get_charger_msg.statu2_1|
-//					s_cancom->get_charger_msg.statu3_1|
-//					s_cancom->get_charger_msg.statu4_1;
+			return s_cancom->get_charger_msg.statu1_1|
+					s_cancom->get_charger_msg.statu2_1|
+					s_cancom->get_charger_msg.statu3_1|
+					s_cancom->get_charger_msg.statu4_1;
 
 
 
@@ -355,11 +409,10 @@ uint8_t hy_can_get_charger_module_statu1(void){
 
 uint8_t hy_can_get_charger_module_statu2(void){
 
-				return s_cancom->get_charger_msg.statu1_2;
-//					|
-//						s_cancom->get_charger_msg.statu2_2|
-//						s_cancom->get_charger_msg.statu3_2|
-//						s_cancom->get_charger_msg.statu4_2;
+				return s_cancom->get_charger_msg.statu1_2|
+						s_cancom->get_charger_msg.statu2_2|
+						s_cancom->get_charger_msg.statu3_2|
+						s_cancom->get_charger_msg.statu4_2;
 	
 
 }
@@ -450,7 +503,7 @@ int hy_can_get_bms_battery_voltage_x10V(void)
 
 int hy_can_set_output_msg(uint16_t voltage_x10V, uint16_t current_x10A)
 {
-	s_cancom->charge_to_bms_msg.output_current_x10A=current_x10A;
+	s_cancom->charge_to_bms_msg.output_current_x10A=current_x10A/4;
 	s_cancom->charge_to_bms_msg.output_voltage_x10V=voltage_x10V;
 	return 0;
 }
diff --git a/User/CAN/hy_can.h b/User/CAN/hy_can.h
index cde1878..966231d 100644
--- a/User/CAN/hy_can.h
+++ b/User/CAN/hy_can.h
@@ -59,30 +59,30 @@ typedef struct CHARGER_MSG{
 	uint32_t battery1_voltage;
 	uint32_t input1_AC_voltage_x10V;
 	
-//	uint32_t output2_current_x10A;
-//	uint32_t output2_voltage_x10V;
-//	uint8_t statu2_1;
-//	uint8_t statu2_2;
-//	uint32_t temperature2_x1degree;	
-//	uint32_t battery2_voltage;
-//	uint32_t input2_AC_voltage_x10V;
-//
-//
-//	uint32_t output3_current_x10A;
-//	uint32_t output3_voltage_x10V;
-//	uint8_t statu3_1;
-//	uint8_t statu3_2;
-//	uint32_t temperature3_x1degree;	
-//	uint32_t battery3_voltage;
-//	uint32_t input3_AC_voltage_x10V;
-//
-//	uint32_t output4_current_x10A;
-//	uint32_t output4_voltage_x10V;
-//	uint8_t statu4_1;
-//	uint8_t statu4_2;
-//	uint32_t temperature4_x1degree;	
-//	uint32_t battery4_voltage;
-//	uint32_t input4_AC_voltage_x10V;
+	uint32_t output2_current_x10A;
+	uint32_t output2_voltage_x10V;
+	uint8_t statu2_1;
+	uint8_t statu2_2;
+	uint32_t temperature2_x1degree;	
+	uint32_t battery2_voltage;
+	uint32_t input2_AC_voltage_x10V;
+
+
+	uint32_t output3_current_x10A;
+	uint32_t output3_voltage_x10V;
+	uint8_t statu3_1;
+	uint8_t statu3_2;
+	uint32_t temperature3_x1degree;	
+	uint32_t battery3_voltage;
+	uint32_t input3_AC_voltage_x10V;
+
+	uint32_t output4_current_x10A;
+	uint32_t output4_voltage_x10V;
+	uint8_t statu4_1;
+	uint8_t statu4_2;
+	uint32_t temperature4_x1degree;	
+	uint32_t battery4_voltage;
+	uint32_t input4_AC_voltage_x10V;
 
 
 	uint32_t setting_voltage_x10V;
diff --git a/User/GUI/GUI_page.c b/User/GUI/GUI_page.c
index af4a0bd..a5938b0 100644
--- a/User/GUI/GUI_page.c
+++ b/User/GUI/GUI_page.c
@@ -309,8 +309,8 @@ PAGE welcomepage(){
     lcd_display_chinese_at(2,1,hysy);
     lcd_display_chinese_at(2,2,anchixinnengyuan);
 	lcd_goto_pos(3,2);
-	lcd_display_num3(48,'V');
-	lcd_display_num3(200,'A');
+	lcd_display_num3(80,'V');
+	lcd_display_num3(150,'A');
 	lcd_display_ascii("-");
 	lcd_display_ascii("qh");
 	hy_gui_delay_ms(2200);
diff --git a/User/Output/hy_output.c b/User/Output/hy_output.c
index ded5f8e..831a861 100644
--- a/User/Output/hy_output.c
+++ b/User/Output/hy_output.c
@@ -86,6 +86,8 @@ int hy_set_charger_output(uint16_t voltage_x10V, uint16_t current_x10A){
 int hy_set_data_broadcast_to_bms(uint16_t voltage_x10V, uint16_t current_x10A)
 {
 	uint8_t status=0;
+	if(voltage_x10V>=960)voltage_x10V=960;
+	if(current_x10A>=1700)current_x10A=1700;
 	hy_can_set_output_msg(voltage_x10V, current_x10A);
 	if(hy_get_charger_module_statu1()&0x80){//硬件故障
 		status |= (1<<0);
